staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -4140,13 +4140,16 @@ func DescribeEvents(el *corev1.EventList, w PrefixWriter) {
 	w.Write(LEVEL_1, "----\t------\t----\t----\t-------\n")
 	for _, e := range el.Items {
 		var interval string
-		if e.Count > 1 {
-			interval = fmt.Sprintf("%s (x%d over %s)", translateTimestampSince(e.LastTimestamp), e.Count, translateTimestampSince(e.FirstTimestamp))
+		firstTimestampSince := translateMicroTimestampSince(e.EventTime)
+		if e.EventTime.IsZero() {
+			firstTimestampSince = translateTimestampSince(e.FirstTimestamp)
+		}
+		if e.Series != nil {
+			interval = fmt.Sprintf("%s (x%d over %s)", translateMicroTimestampSince(e.Series.LastObservedTime), e.Series.Count+1, firstTimestampSince)
```

Comment:
Why `e.Series.Count+1`? Strictly speaking [`e.Series`](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L5545) should be nil when only one event has occurred:

```go
	// Data about the Event series this event represents or nil if it's a singleton Event.
	// +optional
	Series *EventSeries `json:"series,omitempty" protobuf:"bytes,11,opt,name=series"`
```

I can see the argument that since we have an event, `e.Series.Count` should logically be at least 1. However, if the field does contain a number, I don't think we should increment it.

Take a look at this example [from the KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/383-new-event-api-ga-graduation/README.md#short-examples):

```yaml
# After second occurrence, Event looks like:

{
  regarding: A,
  action: B,
  reportingController: C,
  ...,
  series: {count: 2},
}
```

Reply:
I had added the `+1` following a comment in the code:
https://github.com/kubernetes/kubernetes/blob/7a0638da76cb9843def65708b661d2c6aa58ed5a/pkg/printers/internalversion/printers.go#L1789-L1793

Also referred [this](https://github.com/kubernetes/client-go/blob/a31b18a6ac98bbff0dd1055707dfde1b0a98ff68/tools/events/event_broadcaster.go#L183-L186). When a series is created for the first time, its count is set to 1 and for the series to be created, there needs to be a singleton event created.

Reply:
@KnVerey good catch, it looks like at the time the printer PR merged the validation wasn't fully complete. If you check the current code https://github.com/kubernetes/kubernetes/blob/758ad0790ceae9e7be0554637c0fd623ff02e45f/pkg/apis/core/validation/events.go#L110-L117 you'll notice that the minimum required value for counter is 2 and not one like from the above script. Which means that the series represent values exactly like describe in the KEP and not in client-go. So we need to drop `+1` from this PR and also from the printers.